Fhi.Smittestopp.Verification.Tests/Msis/MsisLookupServiceTests.cs

```diff
@@ -0,0 +1,74 @@
+﻿using System;
+using System.Threading.Tasks;
+using Fhi.Smittestopp.Verification.Domain.Models;
+using Fhi.Smittestopp.Verification.Msis;
+using Fhi.Smittestopp.Verification.Msis.Interfaces;
+using Fhi.Smittestopp.Verification.Msis.Models;
+using FluentAssertions;
+using Moq;
+using Moq.AutoMock;
+using NUnit.Framework;
+using Optional;
+
+namespace Fhi.Smittestopp.Verification.Tests.Msis
+{
+    [TestFixture]
+    public class MsisLookupServiceTests
+    {
+        [Test]
+        public async Task FindPositiveTestResult_ResponseNoPositiveTest_ReturnsNone()
+        {
+            var automocker = new AutoMocker();
+
+            automocker.Setup<IMsisClient, Task<Covid19Status>>(x => x.GetCovid19Status("01019012345"))
+                .ReturnsAsync(new Covid19Status
+                {
+                    HarPositivCovid19Prove = false
+                });
+
+            var target = automocker.CreateInstance<MsisLookupService>();
+
+            var result = await target.FindPositiveTestResult("01019012345");
+
+            result.Should().Be(Option.None<PositiveTestResult>());
+        }
+
+        [Test]
+        public async Task FindPositiveTestResult_ResponseHasPositiveTest_ReturnsPositiveTestResult()
+        {
+            var automocker = new AutoMocker();
+
+            var positiveTestData = DateTime.UtcNow.AddDays(-2);
+
+            automocker.Setup<IMsisClient, Task<Covid19Status>>(x => x.GetCovid19Status("01019012345"))
+                .ReturnsAsync(new Covid19Status
+                {
+                    HarPositivCovid19Prove = true,
+                    Provedato = positiveTestData
+                });
+
+            var target = automocker.CreateInstance<MsisLookupService>();
+
+            var result = await target.FindPositiveTestResult("01019012345");
+
+            result.HasValue.Should().BeTrue();
```

Comment:
Assert.Multiple?, ref earlier comment

Reply:
Since we are using FluentAssertions I was under the impression that this was not an option here without mixing in NUnit asserts, but it appears FluentAssertions does have a similar feature with "using (new AssertionScope())", so I will fix this.

Reply:
cool, thumbsup